File: Swift/Two Pointer/1806.swift
var input = readLine()!.split(separator:" ").map{ Int($0)! }; var N = input[0], S = input[1]
var arr = readLine()!.split(separator:" ").map{ Int($0)! }
var L = 0, R = 0
var l = Int.max, ans = arr[0]
while( L <= R && R < N ) {
    if ans >= S {
        ans -= arr[L]
        if l >= (R-L+1) l = (R-L+1)
        L += 1
    }
    else {
        R += 1
        if (R-L+1) >= l {
            ans -= arr[L]
            L += 1
        }
        if R == N break
        ans += arr[R]
    }
}
if l == Int.max print(0)
else print(l)

File: Python/투포인터/1806.py
import sys
input = sys.stdin.readline

N, S = map(int, input().split(" "))
arr = list(map(int, input().split(" ")))
s = arr[0]
ans = float('inf')

if s >= S: print(1)
else:
    left, right = 0, 0
    while( left <= right and right < N):
        if s < S:
            right += 1;
            if right < N:
                s += arr[right]
                if s >= S and ans > right-left+1: ans = right-left+1
        elif s >= S:
            if left < N:
                s -= arr[left]
                left += 1
                if s >= S and ans > right-left+1: ans = right-left+1
    if ans == float('inf'): print(0)
    else: print(ans)
